Николай Кузенцов, домашнее задание 8 спринт

Nikolai-Kuznetsov opened the pull request:
Вопросы ревьюеру:

1. Я поменял название на главном экране на “Playlist Maker”. Можно так оставить? Чтобы оба слова были с большой буквы, как это принято в английском языке? Название на макете: “Playlist Maker”.

2. Я бы на экране настроек сделал бы область клика не только на стрелке назад для перехода на главный экран, но и распространен ещё область клика на слово «Настройки». Например, так реализовано в настройках айфонов.

3. А вот область клика на кнопке «Темная тема» я бы распространил только на switch. При этом этом область клика я бы сделал на всей площади кнопок «Написать в поддержку» и «Пользовательское соглашение». Подобное я заметил на айфоне: область клика ограничивается свитчом, там где он есть.

4. Я бы хотел писать букву «ё», если можно: ) Буква «ё» на макете пишется как «е». Но это уже мелочь. 

5. Я оставил часть кода закоментированным, так как не знаю, нужен ли он будет

Commit:
Init Commit

faritowich commented on app/src/main/java/com/practicum/playlistmaker/MainActivity.kt:
```diff
@@ -9,12 +13,36 @@ import androidx.core.view.WindowInsetsCompat
+//           }
+//       }
+//
+//        buttonSettings.setOnClickListener(clickOnbuttonSettings)
```
Этот закомментированный код можно удалить

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,19 +1,112 @@
+            android:layout_height="0dp"
+            android:layout_weight="1"
+            android:backgroundTint="@color/white"
+            app:cornerRadius="16dp"
```
Повторяющийся размер, в ресурсы

faritowich commented on app/src/main/res/layout/activity_main.xml:
```diff
@@ -1,19 +1,112 @@
+
+            android:foreground="?attr/selectableItemBackground"
+            android:textAllCaps="false"
+
```
Многовато пустых, лучше бы их все убрать

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,163 @@
+            android:textSize="@dimen/text_size_22px"
+            />
+
+    </LinearLayout>
```
Для тулбара можно также использовать androidx.appcompat.widget.Toolbar

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,163 @@
+            android:textColor="@color/black"
+            android:fontFamily="@font/ys_display_regular"
+            android:textSize="@dimen/text_size_16px"
+            android:padding="20dp"
```
Повторяющийся размер

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,163 @@
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:padding="20dp"
+            android:layout_marginEnd="-13dp">
```
Отрицательные отступы тоже лучше не использовать. Не то чтобы это было сильно опасно, просто моветон

faritowich commented on app/src/main/res/layout/activity_settings.xml:
```diff
@@ -0,0 +1,163 @@
+            android:textColor="@color/black"
+            android:fontFamily="@font/ys_display_regular"
+            android:textSize="@dimen/text_size_16px"
+            android:padding="20dp"
```
Лучше такие отступы вынести в родительские лейауты, чтобы не дублировать много раз

Commit:
Вторая версия домашнего задания 8 спринта с правками

Commit:
first commit

Commit:
Вторая версия домашнего задания 8 спринта с правками

Commit:
Переделал шапку в меню настроек. Реализовал androidx.appcompat.widget.Toolbar

Commit:
Отрегулировал цвета главного экрана и кнопок на главном экране.

Внёс изменения в код экрана настроек после того, как переверстал под тул бар.

Commit:
Коммит после отката

Commit:
Решил проблему с текстом в шапке экрана настоек. Настроил ночную и дневную тему шапки, но пока без иконки

Commit:
Merge remote-tracking branch 'origin/dev' into dev

# Conflicts:
#	app/src/main/res/values/styles.xml

Commit:
1. Добавил try-catch в SettingsActivity
2. Перенёс высоту и ширину Тулбара в activity_settings.xml

Commit:
Начал красить стрелку и свитч

Commit:
— Настроил стрелку в тулбаре
— Прописал цвета свитча
— Поправил код в актив сеттингс, убрал ошибки, из-за которых вылетало приложение

Commit:
— Настроил цвета свитча
— Задал настройки текста в тулбаре

Commit:
Создал активити сёрч с шапкой

Commit:
Добавил EditText и новую маленькую иконку лупы

Commit:
Поправил ошибки после добавления EditText и новой маленькой иконки лупы

Commit:
Сверстал окно поиска, создал активити с кодом

Commit:
Сделал курсор синего цвета

Commit:
Написал код для сохранения данных при повороте экрана

Commit:
Причесал код по всему  проекту

Commit:
Перенёс строки SettingsActivity в ресурсы

Commit:
Отсортировал строки SettingsActivity в ресурсы

Commit:
Реализовал переключения ночной темы в приложении
